client/index.ts

```diff
@@ -78,25 +78,77 @@ class OAuth2Client {
   }
 
   private async getJWKS(): Promise<JWKSDT> {
+    if (this.jwks) {
+      return Promise.resolve(this.jwks);
+    } else {
+      return await this.fetchJWKS();
+    }
```

Comment:
As your function is `async` and you don't want to try/catch the `fetchJWKS()` I suggest:

```suggestion
    return this.jwks ? this.jwks : this.fetchJWKS();
```